### PR TITLE
[TypeScript] insomnia-url

### PR DESCRIPTION
**Pull request by dimitropoulos**

### CODE REVIEW

**Comment by dimitropoulos** [comment]
although making separate branches is much nicer for GitHub/review purposes, only those packages that have no other dependencies from the project will work since they consume each others' types. For example, insomnia-inso consumes types from insomnia-testing. If I pull those changes out, it will error because it's expecting types. For that reason I've pushed a branch ts/first7 that contains all of these exact same commits, but stacked. If you want to run something, your best bet will be to do so from there. I'll then make sure that everything is always symmetrical between the two.

File: packages/insomnia-url/yarn.lock [hunk 1]
```diff
@@ -0,0 +1,3642 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (DMarby) [inline packages/insomnia-url/yarn.lock:1]: I'm assuming this is just an accidental leftover?

File: packages/insomnia-url/package.json [hunk 2]
```diff
@@ -4,9 +4,25 @@
   "author": "Gregory Schier <greg.schier@konghq.com>",
   "description": "URL Utilities",
   "license": "MIT",
-  "main": "index.js",
+  "main": "dist/index.js",
+  "types": "dist/index.d.ts",
+  "files": [
+    "dist",
+    "package.json",
+    "READEME.md"
```
Comment (DMarby) [inline packages/insomnia-url/package.json:12]:
```suggestion
    "README.md"
```

File: packages/insomnia-url/src/querystring.ts [hunk 3]
```diff
@@ -64,32 +66,34 @@ module.exports.buildQueryParameter = function(param, strict) {
 
   // Cast number values to strings
   if (typeof param.value === 'number') {
-    param.value += '';
+    param.value = String(param.value);
```
Comment (DMarby) [inline packages/insomnia-url/src/querystring.ts:69]: Nice catch!

**Review by DMarby — APPROVED** [review]

File: packages/insomnia-url/yarn.lock (hunk 1 above)
Comment (dimitropoulos) [reply packages/insomnia-url/yarn.lock:1]: huh.  yes.  shouldn't be there in the first place.  I'll definitely delete.

File: packages/insomnia-url/src/querystring.ts (hunk 3 above)
Comment (dimitropoulos) [reply packages/insomnia-url/src/querystring.ts:69]: Screenshot_20210324_091947

oh, don't thank me.  TypeScript demands this.  both here with the operation, but also in the `flexibleEncodeComponent` function that follows that only accepts `string | undefined` (and, not number).

**Comment by dimitropoulos** [comment]
after feedback (and all fixex therein) and/or approval this PR is closing because this branch is now being incorporated into the `dimitropoulos/typescript` branch.  this isn't final, there is another round of review happening when `dimitropoulos/typescript` is PR'd into `Kong/develop`.

**Pull request closed without merging**